src/A11yDialog.vue

```diff
@@ -1,98 +1,164 @@
-        :aria-labelledby="fullTitleId">
-        <div role="document" :class="classNames.document">
-          <button
-            data-a11y-dialog-hide
-            type="button"
-            :aria-label="closeButtonLabel"
-            @click="close"
-            :class="classNames.closeButton">
-            <slot name="closeButtonContent">
-              {{ '\u00D7' }}
-            </slot>
-          </button>
-
-          <h1 :id="fullTitleId" :class="classNames.title">
-            <slot name="title" />
-          </h1>
-
-          <slot />
-        </div>
-      </component>
+        @click="role === 'alertdialog' ? undefined : close"
+      />
+      <div role="document" :class="classNames.document">
+        <button
+          v-if="closeButtonPosition == 'first'"
+          data-a11y-dialog-hide
+          type="button"
+          :aria-label="closeButtonLabel"
+          :class="classNames.closeButton"
+          @click="close"
+        >
+          <slot name="closeButtonContent">
+            {{ "\u00D7" }}
+          </slot>
+        </button>
+        <p :id="fullTitleId" :class="classNames.title">
+          <slot name="title" />
+        </p>
+        <slot />
+        <button
+          v-if="closeButtonPosition == 'last'"
+          data-a11y-dialog-hide
+          type="button"
+          :aria-label="closeButtonLabel"
+          :class="classNames.closeButton"
+          @click="close"
+        >
+          <slot name="closeButtonContent">
+            {{ "\u00D7" }}
+          </slot>
+        </button>
+      </div>
     </div>
   </teleport>
 </template>
 
 <script>
   import A11yDialog from 'a11y-dialog'
+  import { computed, ref, onMounted, onUnmounted, nextTick } from 'vue'
 
   export default {
-    name: 'VueA11yDialog',
-
+    name: 'A11yDialog',
     props: {
-      id: { type: String, required: true },
-      appRoot: { type: [String, Array], required: true },
-      dialogRoot: { type: String, required: true },
-      classNames: { type: Object, default () { return {} } },
-      titleId: { type: String },
-      closeButtonLabel: { type: String, default: 'Close this dialog window' },
-      disableNative: { type: Boolean, default: false },
-      role: { type: String, default: 'dialog' }
-    },
-
-    computed: {
-      fullTitleId () {
-        return this.titleId || this.id + '-title'
+      id: {
+        type: String,
+        required: true,
       },
-      dialogElement () {
-        return this.disableNative ? 'div' : 'dialog'
+      appRoot: {
+        type: String,
+        required: true,
       },
-      roleAttribute () {
-        return ['dialog', 'alertdialog'].includes(this.role)
-          ? this.role
-          : 'dialog'
+      dialogRoot: {
+        type: String,
+        required: true,
+      },
+      /**
+       * Object representing the classes for each HTML element of the dialog
+       * element. See: https://a11y-dialog.netlify.app/usage/markup
+       */
+      classNames: {
```

Comment:
Just an interesting aside. I was able to utilize this using css modules all the way back from the consumer in AgnosticUI and it totally works since Kitty's a11y-dialog is just writing these out as raw `class=`. I'm a huge fan of css scoping when possible so being able to place CSS Modules e.g. `{ container: this.$style.scopedContainerClass ...` is a huge win I think.

Unfortunately, I could not figure out how to do this in Angular at all and had to turn ViewEncapsulation.None. I think this is a very nice Vue feature (it's of course also available in React but it's very nice that it works in Vue 3 too!)